File: serveur/app/models/muscle.py
# Exemple : app/models/muscle.py

from sqlalchemy import Column, Integer, String, ForeignKey, Boolean, DateTime, func
from app.database import Base

import uuid

class Muscle(Base):
    __tablename__ = "muscles"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    
    name = Column(String, nullable=False)            # exemple: Triceps Long head (niveau précis)
    muscle_group = Column(String, nullable=True)     # exemple: Triceps (niveau intermédiaire)
    zone = Column(String, nullable=True)             # exemple: Arms (niveau haut)
    is_favorite = Column(Boolean, default=False)     # Indique si le muscle est favori ou non
    uuid = Column(String, nullable=False, default=lambda: str(uuid.uuid4()), unique=True)

    updated_at = Column(DateTime(timezone=True), nullable=True)

    def __repr__(self):
        return f"Muscle(id={self.id}, user_id={self.user_id}, name='{self.name}', muscle_group='{self.muscle_group}', zone='{self.zone}', is_favorite={self.is_favorite}, uuid='{self.uuid}', updated_at={self.updated_at})"

File: appli-android/app/src/main/java/com/example/sportapp/core/data/model/Muscle.kt
package com.example.sportapp.core.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.ColumnInfo
import com.example.sportapp.core.utils.CustomDateUtils.getNowISO8601
import java.util.UUID

@Entity(
    tableName = "muscles",
    indices = [androidx.room.Index(value = ["uuid"], unique = true)]
)
data class Muscle(
    @PrimaryKey
    @ColumnInfo(name = "uuid") val uuid: String,

    @ColumnInfo(name = "user_id") val userId: Int,
    val name: String,
    @ColumnInfo(name = "muscle_group") val muscleGroup: String? = null,
    val zone: String? = null,
    @ColumnInfo(name = "is_favorite") val isFavorite: Boolean = false,

    @ColumnInfo(defaultValue = "0") val synced: Boolean = false,
    @ColumnInfo(defaultValue = "0") val pendingDeletion: Boolean = false,

    @ColumnInfo(name = "updated_at") val updatedAt: String? = getNowISO8601(),
)
